ksqldb-common/src/main/java/io/confluent/ksql/util/VertxSslOptionsFactory.java

```diff
@@ -126,6 +154,83 @@ public static Optional<PfxOptions> getPfxTrustStoreOptions(final Map<String, Str
     return Optional.empty();
   }
 
+  /**
+   * Returns a {@code KeyStoreOptions} object using the following truststore SSL configurations:
+   * <ul>
+   *  <li>Optional: {@value SslConfigs#SSL_TRUSTSTORE_PASSWORD_CONFIG}</li>
+   * </ul>
+   *
+   * @param props A Map with the truststore location and password configs.
+   * @return The {@code KeyStoreOptions} configured with the above SSL settings.
+   *         Optional.empty() if the truststore password is null or empty.
+   */
+
+  public static Optional<KeyStoreOptions> getBcfksKeyStoreOptions(final Map<String, String> props) {
+    final String providers = getSecurityProviders(props);
+    final String location = getKeyStoreLocation(props);
+    final String password = getKeyStorePassword(props);
+    final String keyPassword = getKeyPassword(props);
+    final String keyManagerAlgorithm = getKeyManagerAlgorithm(props);
+
+    if (!Strings.isNullOrEmpty(location)
+        && !Strings.isNullOrEmpty(password)
+        && !Strings.isNullOrEmpty(providers)) {
+      return Optional.of(buildBcfksOptions(
+          providers, location, password, keyPassword, keyManagerAlgorithm));
+    }
+
+    return Optional.empty();
+  }
+
+  public static Optional<KeyStoreOptions> getBcfksKeyStoreOptions(
+      final String location, final String password, final String keyPassword) {
+    return getBcfksKeyStoreOptions(ImmutableMap.of(
+        SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, location,
+        SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, password,
+        SslConfigs.SSL_KEY_PASSWORD_CONFIG, keyPassword));
+  }
+
+
+  /**
+   * Returns a {@code KeyStoreOptions} object using the following truststore SSL configurations:
+   * <ul>
+   *  <li>Optional: {@value SslConfigs#SSL_TRUSTSTORE_PASSWORD_CONFIG}</li>
+   * </ul>
+   *
+   * @param props A Map with the truststore location and password configs.
+   * @return The {@code KeyStoreOptions} configured with the above SSL settings.
+   *         Optional.empty() if the truststore password is null or empty.
+   */
+
+  public static Optional<KeyStoreOptions> getBcfksTrustStoreOptions(
+      final Map<String, String> props) {
+    final String providers = getSecurityProviders(props);
+    final String location = getTrustStoreLocation(props);
+    final String password = getTrustStorePassword(props);
+    final String keyPassword = getKeyPassword(props);
+    final String keyManagerAlgorithm = getKeyManagerAlgorithm(props);
+
+    if (!Strings.isNullOrEmpty(location)
+        && !Strings.isNullOrEmpty(password)
+        && !Strings.isNullOrEmpty(providers)) {
+      return Optional.of(buildBcfksOptions(
+          providers, location, password, keyPassword, keyManagerAlgorithm));
+    }
+
+    return Optional.empty();
+  }
+
+  public static Optional<KeyStoreOptions> getBcfksTrustStoreOptions(
+      final String location,
+      final String password,
+      final String keyPassword) {
```

Comment:
is the `keyPassword` needed for truststores? My understanding is that truststores only stores certificates while keystores stores keys. So, truststores don't require a key password, right?